software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java

```diff
@@ -108,23 +108,24 @@ public void start() {
         }
         if (!skipStart) {
             DynamicTasks.queue("install", new Runnable() { public void run() {
-
                 Optional<Boolean> locationInstalled = Optional.fromNullable(getLocation().getConfig(BrooklynConfigKeys.SKIP_ENTITY_INSTALLATION));
                 Optional<Boolean> entityInstalled = Optional.fromNullable(entity.getConfig(BrooklynConfigKeys.SKIP_ENTITY_INSTALLATION));
-                boolean skipInstall = locationInstalled.or(entityInstalled).or(false);
-                if (!skipInstall) {
-                    DynamicTasks.queue("copy-pre-install-resources", new Runnable() { public void run() {
-                        waitForConfigKey(BrooklynConfigKeys.PRE_INSTALL_RESOURCES_LATCH);
-                        copyPreInstallResources();
-                    }});
 
-                    DynamicTasks.queue("pre-install", new Runnable() { public void run() {
-                        preInstall();
-                    }});
+                DynamicTasks.queue("copy-pre-install-resources", new Runnable() { public void run() {
+                    waitForConfigKey(BrooklynConfigKeys.PRE_INSTALL_RESOURCES_LATCH);
+                    copyPreInstallResources();
+                }});
 
-                    DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
-                        runPreInstallCommand();
-                    }});
+                DynamicTasks.queue("pre-install", new Runnable() { public void run() {
+                    preInstall();
+                }});
+
+                DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
+                    runPreInstallCommand();
+                }});
+
+                boolean skipInstall = locationInstalled.or(entityInstalled).or(false);
+                if (!skipInstall) {
```

Comment:
What I really want is a stage that is always executed on entity startup, maybe not `preInstall()` but `prepare()` that doesn't do anything on the VM, but sets up the entity instance. This is where the setting of expandedInstallDir and so on would happen. I'll see if it's easy to make that change.